packages/common/src/ui/layout/tables/columns/EditableQuantityColumn.tsx

```diff
@@ -0,0 +1,73 @@
+import { ColumnDefinition } from './types';
+import { DomainObject } from '../../../../types';
+import { TextField, InputAdornment, Tooltip, Typography } from '@mui/material';
+import React, { useState, useEffect } from 'react';
+
+interface SomeQuantityEntity extends DomainObject {
+  quantity: number;
+  setQuantity: (rowKey: number, newQuantity: number) => void;
+}
+
+export const getEditableQuantityColumn = <
+  T extends SomeQuantityEntity
+>(): ColumnDefinition<T> => ({
+  key: 'quantity',
+  width: 100,
+  Cell: ({ rowData, rowKey }) => {
+    const { quantity } = rowData;
+    const [buffer, setBuffer] = useState(quantity);
+    const [value, setValue] = useState(quantity);
+    const [error, setError] = useState(false);
```

Comment:
all this really depends on what we're doing with the cells and the desired behaviour of what values should / shouldn't be allowed to enter and how restrictive we want to be with actual input.

Ideally I would want to try and avoid excessive constraints (issues with mobile not being able to enter floats because you can enter a `.` for example) and provide feedback instead